.gitlab-ci.py

```diff
@@ -16,13 +17,18 @@
     python.pytest(),
     python.mypy("gcip"),
     python.bdist_wheel(),
+)
+
+dcc = DockerClientConfig(config_file_path="/kaniko/.docker")
+dcc.add_auth("https://index.docker.io/v1/", username_env_var="DOCKER_USER", password_env_var="DOCKER_LOGIN")
```

Comment:
No, I do not think, that Docker Hub should be the default. Any good reasons why this should be handled by us?
I do not want to opinionate the `add_auth()` method.

Reply:
I thought about our discussion if it would be better to default the registry to Docker Hub. If we mention it in the documentation and defaulting it in the function signature it should not be a security issue.

Reply:
Mh, just done some research on api endpoint of index.docker.io. The newer endpoint `/v2` is not supported by kaniko. 
See https://github.com/GoogleContainerTools/kaniko/issues/1209. Workaround would be using `/v1` as stated above, but other tools like crane or dive want to use the newer none deprecated `/v2` endpoint. Should we workaround in `kaniko.execute()` this or shouldn't we preset the registry?

Reply:
I found another challenge. `add_auth` method signature `def add_auth(self, registry: str, username_env_var: str = "REGISTRY_USERNAME", password_env_var: str = "REGISTRY_PASSWORD",) -> None:`, means defaulting to e.g. Docker Hub registry would lead into calling the method without arguments? Like `add_auth()`, if invoked it will create the string: ```json
'echo "{\"auths\": {\"index.docker.com\": {\"username\": \"$REGISTRY_USERNAME\",
  \"password\": \"$REGISTRY_PASSWORD\"}}}" > $HOME/.docker/config.json'
```

Reply:
Suggestion:

- `add_auth(*, registry: Union[Registry, str], ...`
- `Registry` can be a easy to use Enum with well known registry variables (e.g. `DOCKER`, `GCR`, ...)
- `str` allows to pass arbitrary values
- `DOCKER` points to the `/v2` api
- the kaniko job could be that clever that it substitutes any `/v2` docker registry to `/v1`